docs/format.rst

```diff
@@ -330,3 +330,79 @@ The reader load process is as follows:
 2. ``COPY_OBJECT_STORE mydataset.lance/_versions/{version}.manifest-{uuid} mydataset.lance/_versions/{version}.manifest`` reattempt synchronization
 3. ``PUT_EXTERNAL_STORE base_uri, version, mydataset.lance/_versions/{version}.manifest`` update the external store to point to the final manifest
 4. ``RETURN mydataset.lance/_versions/{version}.manifest`` always return the finalized path, return error if synchronization fails
+
+Statistics
+----------
+
+Statistics are stored within Lance files. The statistics can be used to determine
+which pages can be skipped within a query. The null count, lower bound (min),
+and upper bound (max) are stored.
+
+Statistics themselves are stored in Lance's columnar format, which allows for
+selectively reading only relevant stats columns.
+
+Statistic values
+~~~~~~~~~~~~~~~~
+
+Three types of statistics are stored per column: null count, min value, max value.
+The min and max values are stored as their native data types in arrays.
+
+There are special behavior for different data types to account for nulls:
+
+For integer-based data types (including signed and unsigned integers, dates,
+and timestamps), if the min and max are unknown (all values are null), then the
+minimum/maximum representable values should be used instead.
+
+For float data types, if the min and max are unknown, then use ``-Inf`` and ``+Inf``,
```

Comment:
What if the underlying data has `-Inf` or `+Inf`?

Reply:
I imagine the reader wouldn't be affected?

Reply:
Those are known values and can also be recorded as min and/or max.